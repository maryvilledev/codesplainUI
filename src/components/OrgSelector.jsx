import React, { PropTypes } from 'react'
import { SelectField, MenuItem } from 'material-ui'

const isEmpty = (arr) => arr.length === 0

class OrgSelector extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }
  handleChange(ev, tgt, value) {
    const { onChange } = this.props
    onChange(value)
  }
  render() {
    const { value, orgs, style } = this.props;
    return (
      <div>
        <SelectField
          value={value || 0}
          onChange={this.handleChange}
          style={style}
          disabled={!value}
          floatingLabelText='Role'
          >
          {isEmpty(orgs) //No option to set default, so I'll populate with one
                         //instead
            ?
            <MenuItem
              value={0}
              primaryText={"Role"}
            />
            :
            orgs.map((org) => (
            <MenuItem
              key={org}
              value={org}
              primaryText={org}
            />
          ))}
        </SelectField>
      </div>
    );
  }
}

OrgSelector.propTypes = {
  orgs: PropTypes.arrayOf(PropTypes.string).isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  style: PropTypes.object
}

export default OrgSelector;
