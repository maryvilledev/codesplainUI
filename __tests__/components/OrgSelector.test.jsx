import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import OrgSelector from '../../src/components/OrgSelector';

const defaultProps = {
  onChange: jest.fn(),
  orgs: ['galactic-federation', 'international-counsel-of-ricks'],
};

describe('<OrgSelector />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = node => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <OrgSelector {...defaultProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
