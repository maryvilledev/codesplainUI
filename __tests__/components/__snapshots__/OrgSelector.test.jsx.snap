exports[`<OrgSelector /> matches snapshot 1`] = `
<div>
  <SelectField
    autoWidth={false}
    disabled={true}
    floatingLabelText="Role"
    fullWidth={false}
    multiple={false}
    onChange={[Function]}
    value={0}>
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="galactic-federation"
      value="galactic-federation" />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      primaryText="international-counsel-of-ricks"
      value="international-counsel-of-ricks" />
  </SelectField>
</div>
`;
